packages/wallet-management/src/wallets.ts

```diff
@@ -260,12 +260,17 @@ const exodus: Wallet = new InjectedConnector(
 const safe: Wallet = new SafeWalletConnector({
   name: 'Safe',
   installed: async () => {
+    const isIFrameEnvironment = window?.parent !== window;
```

Comment:
We can return earlier if we are not inside the iframe to prevent 99% of our users from waiting for 200ms 🙂

Reply:
Done